test/test-framework/test/e2e/memcached_test.go

```diff
@@ -61,32 +63,47 @@ func memcachedScaleTest(t *testing.T, f *framework.Framework, ctx *framework.Tes
 			Namespace: namespace,
 		},
 		Spec: operator.MemcachedSpec{
-			Size: 3,
+			Size: int32(fromReplicas),
 		},
 	}
 	// use TestCtx's create helper to create the object and add a cleanup function for the new object
 	err = f.Client.Create(goctx.TODO(), exampleMemcached, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
 	if err != nil {
-		return err
+		return fmt.Errorf("could not create example-memcached: %v", err)
 	}
-	// wait for example-memcached to reach 3 replicas
-	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "example-memcached", 3, retryInterval, timeout)
+	// wait for example-memcached to reach `fromReplicas` replicas
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "example-memcached", fromReplicas, retryInterval, timeout)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed waiting for %d example-memcached replicas: %v", fromReplicas, err)
 	}
 
-	err = f.Client.Get(goctx.TODO(), types.NamespacedName{Name: "example-memcached", Namespace: namespace}, exampleMemcached)
-	if err != nil {
-		return err
-	}
-	exampleMemcached.Spec.Size = 4
-	err = f.Client.Update(goctx.TODO(), exampleMemcached)
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
You way want to just use this:
https://godoc.org/k8s.io/client-go/util/retry#RetryOnConflict

Reply:
I would. Forget that existed. Thanks.